src/encoded/views/download.py

```diff
@@ -0,0 +1,154 @@
+    ('Biosample treatments', ('replicates.library.biosample.treatments.treatment_term_name')),
+    ('Biosample subcellular fraction term name', ('replicates.library.biosample.subcellular_fraction_term_name')),
+    ('Biosample phase', ('replicates.library.biosample.phase')),
+    ('Experiment target', 'target.name'),
+    ('Antibody accession', ('replicates.antibody.accession')),
+    ('Nucleic acid term name', ('replicates.library.nucleic_acid_term_name')),
+    ('Depleted in term name', ('replicates.library.depleted_in_term_name')),
+    ('Library extraction method', ('replicates.library.extraction_method')),
+    ('Library fragmentation method', ('replicates.library.fragmentation_method')),
+    ('Libary lysis method', ('replicates.library.lysis_method')),
+    ('Read length', ('replicates.library.read_length')),
+    ('Paired endedness', ('replicates.library.paired_ended')),
+    ('Experiment date released', ('date_released')),
+    ('Project', ('award.project')),
+    ('File format', ('files.file_format')),
+    ('Size', ('files.file_size')),
+    ('Output type', ('files.output_type')),
+    ('Lab', ('files.lab')),
+    ('md5sum', ('files.md5sum')),
+    ('href', ('files.href')),
+    ('Assembly', ('files.assembly'))
+])
+
+
+def dict_generator(indict, pre=None):
+    pre = pre[:] if pre else []
+    if isinstance(indict, dict):
+        for key, value in indict.items():
+            if isinstance(value, dict):
+                for d in dict_generator(value, [key] + pre):
+                    yield d
+            elif isinstance(value, list) or isinstance(value, tuple):
+                for v in value:
+                    for d in dict_generator(v, [key] + pre):
+                        yield d
+            else:
+                yield pre + [key, value]
+    else:
+        yield indict
+
+
+@view_config(route_name='metadata', request_method='GET', renderer='tsv')
+def metadata_tsv(context, request):
+
+    param_list = parse_qs(request.matchdict['search_params'].encode('utf-8'))
+    param_list['field'] = []
+    header = []
+    file_attributes = []
+    for prop in _tsv_mapping:
+        header.append(prop)
+        param_list['field'] = param_list['field'] + [_tsv_mapping[prop]]
+        if _tsv_mapping[prop].startswith('files'):
+            file_attributes.append(_tsv_mapping[prop])
+    param_list['limit'] = ['all']
+    path = '/search/?%s' % urlencode(param_list, True)
+    results = embed(request, path, as_user=True)
+    rows = []
+    for row in results['@graph']:
+        if row['files']:
+            exp_data_row = []
+            new_data = []
+            for arr in dict_generator(row):
+                new_data.append(arr)
+            for column in header:
+                if not _tsv_mapping[column].startswith('files'):
+                    path = _tsv_mapping[column].split('.')
+                    temp = []
+                    for ld in new_data:
+                        if sorted(path) == sorted(ld[:-1]):
```

Comment:
Yeah I have to modify

```
dict_generator
```

to have paths in order and remove sorting but as of now I don't have any conflicts.
Can you think of a path which will be a problem